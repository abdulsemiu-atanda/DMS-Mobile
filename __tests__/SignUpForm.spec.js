import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import SignUpForm from '../components/SignUpForm.react'
import {signUpFixtures} from '../util/fixtures/formFixtures'

describe('SignUpForm', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SignUpForm {...signUpFixtures()} />).toJSON()

    expect(tree).toMatchSnapshot()
    expect(tree.children).toHaveLength(12)
  })

  it('render correctly when rendering conditions are present', () => {
    const tree = renderer.create(
      <SignUpForm {...signUpFixtures()} passwordMatch='false' signUpFail />
    ).toJSON()

    expect(tree).toMatchSnapshot()
    expect(tree.children).toHaveLength(14)
  })
})
