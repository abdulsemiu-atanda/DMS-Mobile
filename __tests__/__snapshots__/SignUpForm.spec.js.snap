// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm render correctly when rendering conditions are present 1`] = `
<View
  style={
    Object {
      "flex": 2,
      "padding": 30,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "fontSize": 18,
        "marginBottom": 20,
        "textAlign": "center",
      }
    }
  >
    Sign Up
  </Text>
  <Text
    style={
      Object {
        "color": "#96a1ae",
        "fontSize": 10,
      }
    }
  >
    FULL NAME
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    placeholder="Jason Bourne"
    style={
      Object {
        "borderBottomColor": "#bbb",
        "borderBottomWidth": 1,
        "marginBottom": 20,
        "marginTop": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Text
    style={
      Object {
        "color": "#96a1ae",
        "fontSize": 10,
      }
    }
  >
    USERNAME
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    placeholder="jbourne"
    style={
      Object {
        "borderBottomColor": "#bbb",
        "borderBottomWidth": 1,
        "marginBottom": 20,
        "marginTop": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Text
    style={
      Object {
        "color": "#96a1ae",
        "fontSize": 10,
      }
    }
  >
    EMAIL
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    keyboardType="email-address"
    placeholder="jason@bourne.com"
    style={
      Object {
        "borderBottomColor": "#bbb",
        "borderBottomWidth": 1,
        "marginBottom": 20,
        "marginTop": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Text
    style={
      Object {
        "color": "#96a1ae",
        "fontSize": 10,
      }
    }
  >
    PASSWORD
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    secureTextEntry={true}
    style={
      Object {
        "borderBottomColor": "#bbb",
        "borderBottomWidth": 1,
        "marginBottom": 20,
        "marginTop": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Text
    style={
      Object {
        "color": "#96a1ae",
        "fontSize": 10,
      }
    }
  >
    CONFIRM PASSWORD
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onEndEditing={[Function]}
    secureTextEntry={true}
    style={
      Object {
        "borderBottomColor": "#bbb",
        "borderBottomWidth": 1,
        "marginBottom": 20,
        "marginTop": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0983ff",
        "height": 40,
        "marginTop": 20,
        "padding": 7,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#aacfff",
          "fontSize": 18,
        }
      }
    >
      Create
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "red",
        "marginTop": 10,
      }
    }
  >
    Your password do not match
  </Text>
  <Text
    style={
      Object {
        "color": "red",
        "marginTop": 10,
      }
    }
  />
</View>
`;

exports[`SignUpForm renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 2,
      "padding": 30,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "fontSize": 18,
        "marginBottom": 20,
        "textAlign": "center",
      }
    }
  >
    Sign Up
  </Text>
  <Text
    style={
      Object {
        "color": "#96a1ae",
        "fontSize": 10,
      }
    }
  >
    FULL NAME
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    placeholder="Jason Bourne"
    style={
      Object {
        "borderBottomColor": "#bbb",
        "borderBottomWidth": 1,
        "marginBottom": 20,
        "marginTop": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Text
    style={
      Object {
        "color": "#96a1ae",
        "fontSize": 10,
      }
    }
  >
    USERNAME
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    placeholder="jbourne"
    style={
      Object {
        "borderBottomColor": "#bbb",
        "borderBottomWidth": 1,
        "marginBottom": 20,
        "marginTop": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Text
    style={
      Object {
        "color": "#96a1ae",
        "fontSize": 10,
      }
    }
  >
    EMAIL
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    keyboardType="email-address"
    placeholder="jason@bourne.com"
    style={
      Object {
        "borderBottomColor": "#bbb",
        "borderBottomWidth": 1,
        "marginBottom": 20,
        "marginTop": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Text
    style={
      Object {
        "color": "#96a1ae",
        "fontSize": 10,
      }
    }
  >
    PASSWORD
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    secureTextEntry={true}
    style={
      Object {
        "borderBottomColor": "#bbb",
        "borderBottomWidth": 1,
        "marginBottom": 20,
        "marginTop": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Text
    style={
      Object {
        "color": "#96a1ae",
        "fontSize": 10,
      }
    }
  >
    CONFIRM PASSWORD
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    onEndEditing={[Function]}
    secureTextEntry={true}
    style={
      Object {
        "borderBottomColor": "#bbb",
        "borderBottomWidth": 1,
        "marginBottom": 20,
        "marginTop": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0983ff",
        "height": 40,
        "marginTop": 20,
        "padding": 7,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#aacfff",
          "fontSize": 18,
        }
      }
    >
      Create
    </Text>
  </View>
</View>
`;
